nibabel/__init__.py

```diff
@@ -35,16 +35,28 @@
 For more detailed information see the :ref:`manual`.
 """
 
-
-def setup_test():
-    """ Set numpy print options to "legacy" for new versions of numpy
-
-    If imported into a file, nosetest will run this before any doctests.
-    """
-    import numpy
+# Package-wide test setup and teardown
+# Numpy changed print options in 1.14; we can update docstrings and remove
+# these when our minimum for building docs exceeds that
+_save_printopts = None
+
+def setup_package():
+    """ Set numpy print style to legacy="1.13" for newer versions of numpy """
+    import nibabel as nb
+    import numpy as np
     from distutils.version import LooseVersion
-    if LooseVersion(numpy.__version__) >= LooseVersion('1.14'):
-        numpy.set_printoptions(legacy="1.13")
+    if nb._save_printopts is None:
+        nb._save_printopts = np.get_printoptions().get('legacy')
```

Comment:
so it is not "printopts" really but a "legacyprintopt", right?

FWIW -- ATM it is just a single variable here. In DataLad we setup/teardown a few, so I just made a `_test_states` dict variable: https://github.com/datalad/datalad/blob/master/datalad/__init__.py#L92 to store all those in without breeding them at the top module level.